### ISSUE TITLE
Rule proposal: Forbid nested assertions

### ISSUE THREAD

**Issue by jfmengels** [avajs/eslint-plugin-ava#150]
I have just received a PR where the contributor added a test resembling this:

``` js
test('should foo...', t => {
  t.is(
     t.throws(() => {
       configFile.readConfig(absentFile);
     }).message,
     'Could not find X'
   );
});
```

This could obviously be simplified to 

``` js
t.throws(() => {
  configFile.readConfig(absentFile);
}, 'Could not find X');
```

I think this could have been prevented by telling the user not to use an assertion inside another assertion, though I'm not sure whether the error message would have helped the user know how else to do this.

**Comment by novemberborn** [comment]
Could make it specific to `t.throws`, and say "assign t.throws() result to a variable before combining it with another assertion", or something like that?

**Comment by jfmengels** [comment]
True, the message would be a lot nicer then (though even with a more generic rule, we can still have different error messages).

Unless someone has other ideas of how assertions could be nested together except with `t.throws`, I'm leaning towards what @novemberborn said, having one rule for this specific case.

Actually, I do have a different case in mind, but also dealing with `t.throws`:
```js
test('foo', t => {
  t.throws(() => {
    t.is(1, 2);
  });
});
```
